src/gc/gcinterface.h

```diff
@@ -0,0 +1,499 @@
+    virtual BOOL RegisterForFullGCNotification(uint32_t gen2Percentage, uint32_t lohPercentage) = 0;
+
+    // Cancels a full GC notification that was requested by `RegisterForFullGCNotification`.
+    virtual BOOL CancelFullGCNotification() = 0;
+
+    // Returns the status of a registered notification for determining whether a blocking
+    // Gen 2 collection is about to be initiated, with the given timeout.
+    virtual int WaitForFullGCApproach(int millisecondsTimeout) = 0;
+
+    // Returns the status of a registered notification for determining whether a blocking
+    // Gen 2 collection has completed, with the given timeout.
+    virtual int WaitForFullGCComplete(int millisecondsTimeout) = 0;
+
+    // Returns the generation in which obj is found.
+    virtual unsigned WhichGeneration(Object* obj) = 0;
+
+    // Returns the number of GCs that have transpired in the given generation
+    // since the beginning of the life of the process.
+    virtual int CollectionCount(int generation, int get_bgc_fgc_coutn = 0) = 0;
+
+    // Begins a no-GC region, returning a code indicating whether entering the no-GC
+    // region was successful.
+    virtual int StartNoGCRegion(uint64_t totalSize, BOOL lohSizeKnown, uint64_t lohSize, BOOL disallowFullBlockingGC) = 0;
+
+    // Exits a no-GC region.
+    virtual int EndNoGCRegion() = 0;
+
+    // Gets the total number of bytes in use.
+    virtual size_t GetTotalBytesInUse() = 0;
+
+    // Forces a garbage collection of the given generation.
+    virtual HRESULT GarbageCollect(int generation = -1, BOOL low_memory_p = FALSE, int mode = collection_blocking) = 0;
+
+    // Gets the largest GC generation.
+    virtual unsigned GetMaxGeneration() = 0;
+
+    // Indicates that an object's finalizer should not be run upon the object's collection.
+    virtual void SetFinalizationRun(Object* obj) = 0;
+
+    // Indicates that an object's finalizer should be run upon the object's collection.
+    virtual bool RegisterForFinalization(int gen, Object* obj) = 0;
+
+    /*
+    ===========================================================================
+    Miscellaneous routines used by the VM.
+    ===========================================================================
+    */
+
+    // Initializes the GC heap, returning whether or not the initialization
+    // was successful.
+    virtual HRESULT Initialize() = 0;
+
+    // Returns whether nor this GC was promoted by the last GC.
+    virtual BOOL IsPromoted(Object* object) = 0;
+
+    // Returns true if this pointer points into a GC heap, false otherwise.
+    virtual BOOL IsHeapPointer(void* object, BOOL small_heap_only = FALSE) = 0;
+
+    // Return the generation that has been condemned by the current GC.
+    virtual unsigned GetCondemnedGeneration() = 0;
+
+    // Get the timestamp corresponding to the last GC that occured for the
+    // given generation.
+    virtual size_t GetLastGCStartTime(int generation) = 0;
```

Comment:
>GetLastGCStartTime [](start = 19, length = 18)

should make a note to move these 3 APIs off of the interface as they are only used by the Add/RemoveMemoryPressure which is to be moved to the GC side.